Build failure: shairport-sync

### Steps To Reproduce

Steps to reproduce the behavior:

1. Make `configuration.nix` as such, so that it builds with Airplay 2 support:

  ```
    nixpkgs.overlays = [
     (final: prev: { shairport-sync = prev.shairport-sync.override { enableAirplay2 = true; }; })
    ];
  
    systemd.services."nqptp" = {
      description = "nqptp";
      enable = true;
      serviceConfig = {
        ExecStart = "${pkgs.nqptp}/bin/nqptp";
        #Restart = "always";
      };
      wantedBy = [ "default.target" ];
    };
  
    services.shairport-sync = {
      enable = true;
      arguments = "--name=Foo --verbose --output=alsa -- -d hw:Whatever";
    };
  ```

2. `nixos-rebuild switch`

### Build log

```
error: build of '/nix/store/p8y9r3vih4nd432k37hbimab2n8fj4mj-shairport-sync-4.1.1.drv' on 'ssh-ng://builder' failed: builder for '/nix/store/p8y9r3vih4nd432k37hbimab2n8fj4mj-shairport-sync-4.1.1.drv' failed with exit code 1;
       last 10 log lines:
       > checking for soxr... yes
       > checking for avahi_client_new in -lavahi-client... yes
       > checking for avahi_strerror in -lavahi-common... yes
       > checking for alsa... yes
       > checking for jack... yes
       > checking for libpulse >= 0.9.2... yes
       > checking for libpipewire-0.3 >= 0.3.24... yes
       > checking for gio-unix-2.0 >= 2.30.0... yes
       > checking for xxd... no
       > configure: error: xxd can not be found. Please install xxd for building for AirPlay 2.
       For full logs, run 'nix-store -l /nix/store/p8y9r3vih4nd432k37hbimab2n8fj4mj-shairport-sync-4.1.1.drv'.
```

### Additional context

Strangest thing is that I have 4.1.1 built and working for a while (probably at 9722f8221842b409606f7d0af80fe2592f374071) but following a `nix-channel --update` it does not rebuild anymore, even though [`xxd` is listed in](https://github.com/NixOS/nixpkgs/blob/32dcb45f66c0487e92db8303a798ebc548cadedc/pkgs/servers/shairport-sync/default.nix#L80-L87) and hasn't changed, while the only two subsequent commits are 4b939b45c6911216c441761b714e8a0d49af54e7 and fe6662f8d48779b6b82cacb0f196039fb0a737a6 which I'm not too sure how they could relate.

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@LnL7 @jordanisaacs 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-linux"`
 - host os: `Linux 6.1.21, NixOS, 23.05 (Stoat), 23.05.1156.ad157fe26e7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"nixos-23.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/servers/shairport-sync/default.nix
+++ b/pkgs/servers/shairport-sync/default.nix
@@ -58,6 +58,8 @@ stdenv.mkDerivation rec {
     # To achieve this, we coerce the output to a string to prevent
     # mkDerivation's splicing logic from kicking in.
     "${glib.dev}"
+  ] ++ optional enableAirplay2 [
+    unixtools.xxd
   ];
 
   makeFlags = [
@@ -83,7 +85,6 @@ stdenv.mkDerivation rec {
     libgcrypt
     libuuid
     ffmpeg
-    unixtools.xxd
   ]
   ++ optional stdenv.isLinux glib;
 

